### PR TITLE
remove invalidate parenthesis

### PR DESCRIPTION
**Pull request by wilsonyqm**

### CODE REVIEW

**Comment by wilsonyqm** [comment]
[Remove Invalid Parenthesis](https://trello.com/c/EKrGDixd/11-remove-invalid-parenthesis)

File: bfs/removeInvalidParenthesis.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
+//BFS is definitely better than dfs because it only cares about the max result.
```
Comment (Ethan199111) [inline bfs/removeInvalidParenthesis.java:1]: Exactly, avoid using DFS. No industry code is using recursive solution.

**Review by Ethan199111 — APPROVED** [review]

File: bfs/removeInvalidParenthesis.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+                            visited.add(newStr);
+                        }
+                    }
+                }
+            }
+        }
+        return res;
+    }
+
+    public boolean isValid(char[] arr) {
```
Comment (Ethan199111) [inline bfs/removeInvalidParenthesis.java:35]: this can be static

Comment (wilsonyqm) [reply bfs/removeInvalidParenthesis.java:35]: Why?

Comment (Ethan199111) [reply bfs/removeInvalidParenthesis.java:35]: Actually, the whole class can be static. So you don't need to create a instance to call its methods.

Comment (wilsonyqm) [reply bfs/removeInvalidParenthesis.java:35]: .. Come on Xiaoge You are so into static class

Comment (wilsonyqm) [reply bfs/removeInvalidParenthesis.java:35]: https://stackoverflow.com/a/5313383

**Pull request merged**